lib/src/screens/buy/buy_sell_page.dart

```diff
@@ -0,0 +1,471 @@
+        _fiatAmountDebounce.run(() {
+          buySellViewModel.changeFiatAmount(amount: fiatAmountController.text);
+        });
+      }
+    });
+
+    cryptoAmountController.addListener(() {
+      if (cryptoAmountController.text != buySellViewModel.cryptoAmount) {
+        _cryptoAmountDebounce.run(() {
+          buySellViewModel.changeCryptoAmount(amount: cryptoAmountController.text);
+        });
+      }
+    });
+
+    cryptoAddressController.addListener(() {
+      buySellViewModel.changeCryptoCurrencyAddress(cryptoAddressController.text);
+    });
+
+
+    _isReactionsSet = true;
+  }
+
+  void _onCryptoCurrencyChange(CryptoCurrency currency, BuySellViewModel buySellViewModel,
+      GlobalKey<ExchangeCardState> key) {
+    final isCurrentTypeWallet = currency == buySellViewModel.wallet.currency;
+
+    key.currentState!.changeSelectedCurrency(currency);
+
+    key.currentState!.changeAddress(
+        address: isCurrentTypeWallet ? buySellViewModel.wallet.walletAddresses.address : '');
+
+    key.currentState!.changeAmount(amount: '');
+  }
+
+  void _onFiatCurrencyChange(
+      FiatCurrency currency, BuySellViewModel buySellViewModel, GlobalKey<ExchangeCardState> key) {
+    key.currentState!.changeSelectedCurrency(currency);
+    key.currentState!.changeAmount(amount: '');
+  }
+
+  void disposeBestRateSync() => {};
+
+  Widget _exchangeCardsSection(BuildContext context) {
+    final fiatExchangeCard = Observer(
+        builder: (_) => ExchangeCard(
+              onDispose: disposeBestRateSync,
+              amountFocusNode: _fiatAmountFocus,
+              addressFocusNode: _depositAddressFocus,
+              key: fiatCurrencyKey,
+              title: 'FIAT ${S.of(context).amount}',
+              initialCurrency: buySellViewModel.fiatCurrency,
+              initialWalletName: '',
+              initialAddress: '',
+              initialIsAmountEditable: true,
+              isAmountEstimated: false,
+              currencyRowPadding: EdgeInsets.zero,
+              addressRowPadding: EdgeInsets.zero,
+              isMoneroWallet: buySellViewModel.wallet == WalletType.monero,
+              showAddressField: false,
+              showLimitsField: false,
+              currencies: buySellViewModel.fiatCurrencies,
+              onCurrencySelected: (currency) =>
+                  buySellViewModel.changeFiatCurrency(currency: currency),
+              imageArrow: arrowBottomPurple,
+              currencyButtonColor: Colors.transparent,
+              addressButtonsColor:
+                  Theme.of(context).extension<SendPageTheme>()!.textFieldButtonColor,
+              borderColor:
+                  Theme.of(context).extension<ExchangePageTheme>()!.textFieldBorderTopPanelColor,
+              currencyValueValidator: (value) {
+                return null;
+              },
+              addressTextFieldValidator: AddressValidator(type: buySellViewModel.cryptoCurrency),
```

Comment:
**Minor**
this is the fiat amount, why address validator?

Reply:
removed